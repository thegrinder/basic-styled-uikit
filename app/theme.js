const theme = {
  uiKit: {
    buttons: {
      btnTypes: {
        default: {
          regular: {
            color: '#222',
            bgColor: 'transparent',
            borderColor: '#e5e5e5',
          },
          hover: {
            color: '#222',
            bgColor: 'transparent',
            borderColor: '#b2b2b2',
          },
          pressed: {
            color: '#222',
            bgColor: 'transparent',
            borderColor: '#999',
          },
        },
        primary: {
          regular: {
            color: '#fff',
            bgColor: '#1e87f0',
            borderColor: 'transparent',
          },
          hover: {
            color: '#fff',
            bgColor: '#0f7ae5',
            borderColor: 'transparent',
          },
          pressed: {
            color: '#fff',
            bgColor: '#0e6dcd',
            borderColor: 'transparent',
          },
        },
        secondary: {
          regular: {
            color: '#fff',
            bgColor: '#222',
            borderColor: 'transparent',
          },
          hover: {
            color: '#fff',
            bgColor: '#151515',
            borderColor: 'transparent',
          },
          pressed: {
            color: '#fff',
            bgColor: '#080808',
            borderColor: 'transparent',
          },
        },
        danger: {
          regular: {
            color: '#fff',
            bgColor: '#f0506e',
            borderColor: 'transparent',
          },
          hover: {
            color: '#fff',
            bgColor: '#ee395b',
            borderColor: 'transparent',
          },
          pressed: {
            color: '#fff',
            bgColor: '#ec2147',
            borderColor: 'transparent',
          },
        },
      },
      btnSizes: {
        small: {
          padding: '0 15px',
          lineHeight: '28px',
          fontSize: '14px',
        },
        default: {
          padding: '0 30px',
          fontSize: '14px',
          lineHeight: '38px',
        },
        large: {
          padding: '0 40px',
          lineHeight: '53px',
          fontSize: '16px',
        },
      },
    },
  },
};

export default theme;
