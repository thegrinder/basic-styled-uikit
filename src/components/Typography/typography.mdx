---
name: Typography
menu: Components
route: /components/typography
---

import { Playground, Props } from 'docz';
import { BoxProvider, Box } from '../Box';
import TypographyProvider from './TypographyProvider';
import Text from './Text/Text';
import Heading from './Heading/Heading';


## **TypographyProvider**

`<TypographyProvider />` is responsible for passing down the theme to the `<Heading />` and `<Text />` component. It expects the theme to be of certain structure.


---


## **Heading**

Heading component grabs the theme given to `<TypographyProvider />` and applies the style defined there based on the given `color`, `sizing`, and `emphasis` prop values. It also accepts a few different non-theme related props for convenience reasons.

<Props of={Heading} />


---


## **Text**

Text component grabs the theme given to `<TypographyProvider />` and applies the style defined there based on the given `color`, `sizing`, and `emphasis` prop values. It also accepts a few different non-theme related props for convenience reasons.

<Props of={Text} />


---


### Example


<Playground>
  {() => {
    const boxTheme = {
      bgColors: {
        neutral: '#fff',
        primary: '#1e88e5',
        success: '#43a047',
        warning: '#fdd835',
        danger: '#f4511e',
      },
    };

    const typographyTheme = {
      baseFontSize: 16,
      baseLineHeight: 1.5,
      opacities: {
        high: 1,
        medium: 0.8,
        low: 0.6,
      },
      colors: {
        neutral: '#161616',
        primary: '#1e88e5',
        success: '#43a047',
        warning: '#fdd835',
        danger: '#f4511e',
      },
      onBackground: {
        primary: {
          neutral: '#fff',
        },
        success: {
          neutral: '#fff',
        },
        warning: {
          neutral: '#161616',
        },
        danger: {
          neutral: '#fff',
        },
      },
      text: {
        sizings: {
          l: {
            fontSize: '1.25rem',
          },
          m: {
            fontSize: '1rem',
          },
          s: {
            fontSize: '.875rem',
          },
          xs: {
            fontSize: '.75rem',
          },
        },
        common: {
          fontFamily: `'Inter var', sans-serif`,
        },
      },
      heading: {
        sizings: {
          h1: {
            fontSize: '3rem',
          },
          h2: {
            fontSize: '2.5rem',
          },
          h3: {
            fontSize: '2rem',
          },
          h4: {
            fontSize: '1.5rem',
          },
          h5: {
            fontSize: '1.25rem',
          },
          h6: {
            fontSize: '1rem',
          },
        },
        common: {
          fontFamily: `'Inter var', sans-serif`,
          fontWeight: 600,
        },
      },
    };

    const Example = () => (
      <BoxProvider theme={boxTheme}>
        <TypographyProvider theme={typographyTheme}>
          <>
            <h2>Heading sizes:</h2>
            <Heading as="h1">h1 heading</Heading>
            <Heading as="h2">h2 heading</Heading>
            <Heading as="h3">h3 heading</Heading>
            <Heading as="h4">h4 heading</Heading>
            <Heading as="h5">h5 heading</Heading>
            <Heading as="h6">h6 heading</Heading>
            <Heading sizing="h6" as="h1" marginBottom>
              small h1 heading
            </Heading>

            <h2>Heading colors:</h2>
            <Heading as="h6" color="neutral">neutral heading</Heading>
            <Heading as="h6" color="primary">primary heading</Heading>
            <Heading as="h6" color="danger">danger heading</Heading>
            <Heading as="h6" color="warning">warning heading</Heading>
            <Heading as="h6" color="success" marginBottom>
              success heading
            </Heading>

            <h2>Heading colors on different backgrounds:</h2>
            <Box bgColor="neutral">
              <Heading as="h6">heading on neutral backgroud</Heading>
            </Box>
            <Box bgColor="primary">
              <Heading as="h6">heading on primary backgroud</Heading>
            </Box>
            <Box bgColor="success">
              <Heading as="h6">heading on success backgroud</Heading>
            </Box>
            <Box bgColor="warning">
              <Heading as="h6">heading on warning backgroud</Heading>
            </Box>
            <Box bgColor="danger" className="margin-bottom">
              <Heading as="h6">heading on danger backgroud</Heading>
            </Box>

            <h2>Text sizes:</h2>
            <Text sizing="l">large text</Text>
            <Text sizing="m">medium text</Text>
            <Text sizing="s">small text</Text>
            <Text sizing="xs" marginBottom>
              extra small text
            </Text>

            <h2>Text colors:</h2>
            <Text color="neutral">neutral text</Text>
            <Text color="primary">primary text</Text>
            <Text color="danger">danger text</Text>
            <Text color="warning">warning text</Text>
            <Text color="success">success text</Text>

            <h2>Text colors on different backgrounds:</h2>
            <Box bgColor="neutral">
              <Text>text on neutral backgroud</Text>
            </Box>
            <Box bgColor="primary">
              <Text>text on primary backgroud</Text>
            </Box>
            <Box bgColor="success">
              <Text>text on success backgroud</Text>
            </Box>
            <Box bgColor="warning">
              <Text>text on warning backgroud</Text>
            </Box>
            <Box bgColor="danger">
              <Text>text on danger backgroud</Text>
            </Box>

          </>
        </TypographyProvider>
      </BoxProvider>
    );

    return <Example />;
  }}
</Playground>
