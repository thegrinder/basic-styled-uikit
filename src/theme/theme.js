import button from '../components/Button/buttonTheme';
import link from '../components/Link/linkTheme';
import form from '../components/Form/formTheme';
import text from '../components/Text/textTheme';
import heading from '../components/Headings/headingTheme';
import colorBox from '../components/ColorBox/colorBoxTheme';

const theme = {
  uiKit: {
    button,
    link,
    form,
    text,
    heading,
    colorBox,
  },
};

export default theme;
