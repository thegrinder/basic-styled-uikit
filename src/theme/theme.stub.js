import { rem } from 'basic-styled-uikit';

const colors = {
  transparent: 'transparent',
  white: '#fff',
  gray50: '#fafafa',
  gray100: '#f8f9fa',
  gray200: '#e9ecef',
  gray300: '#dee2e6',
  gray400: '#ced4da',
  gray500: '#adb5bd',
  gray600: '#868e96',
  gray700: '#495057',
  gray800: '#343a40',
  gray900: '#212529',
  black: '#161616',

  primary50: '#e3f2fd',
  primary100: '#bbdefb',
  primary200: '#90caf9',
  primary300: '#64b5f6',
  primary400: '#42a5f5',
  primary500: '#2196f3',
  primary600: '#1e88e5',
  primary700: '#1976d2',
  primary800: '#1565c0',
  primary900: '#0d47a1',

  success50: '#e8f5e9',
  success100: '#c8e6c9',
  success200: '#a5d6a7',
  success300: '#81c784',
  success400: '#66bb6a',
  success500: '#4caf50',
  success600: '#43a047',
  success700: '#388e3c',
  success800: '#2e7d32',
  success900: '#1b5e20',

  warning50: '#fffde7',
  warning100: '#fff9c4',
  warning200: '#fff59d',
  warning300: '#fff176',
  warning400: '#ffee58',
  warning500: '#ffeb3b',
  warning600: '#fdd835',
  warning700: '#fbc02d',
  warning800: '#f9a825',
  warning900: '#f57f17',

  danger50: '#ffebee',
  danger100: '#ffcdd2',
  danger200: '#ef9a9a',
  danger300: '#e57373',
  danger400: '#ef5350',
  danger500: '#f44336',
  danger600: '#e53935',
  danger700: '#d32f2f',
  danger800: '#c62828',
  danger900: '#b71c1c',
};

const theme = {
  uiKit: {
    colors,
    // button theme
    button: {
      btnTypes: {
        basic: {
          normal: {
            color: colors.gray900,
            backgroundColor: colors.transparent,
            borderColor: colors.gray300,
          },
          hover: {
            color: colors.gray900,
            backgroundColor: colors.transparent,
            borderColor: colors.gray400,
          },
          active: {
            color: colors.gray900,
            backgroundColor: colors.transparent,
            borderColor: colors.gray500,
          },
        },
        default: {
          normal: {
            color: colors.black,
            backgroundColor: colors.gray300,
            borderColor: colors.transparent,
          },
          hover: {
            color: colors.black,
            backgroundColor: colors.gray400,
            borderColor: colors.transparent,
          },
          active: {
            color: colors.black,
            backgroundColor: colors.gray500,
            borderColor: colors.transparent,
          },
        },
        primary: {
          normal: {
            color: colors.white,
            backgroundColor: colors.primary500,
            borderColor: colors.transparent,
          },
          hover: {
            color: colors.white,
            backgroundColor: colors.primary600,
            borderColor: colors.transparent,
          },
          active: {
            color: colors.white,
            backgroundColor: colors.primary700,
            borderColor: colors.transparent,
          },
        },
        danger: {
          normal: {
            color: colors.white,
            backgroundColor: colors.danger500,
            borderColor: colors.transparent,
          },
          hover: {
            color: colors.white,
            backgroundColor: colors.danger600,
            borderColor: colors.transparent,
          },
          active: {
            color: colors.white,
            backgroundColor: colors.danger700,
            borderColor: colors.transparent,
          },
        },
      },
      btnSizes: {
        small: {
          padding: `0 ${rem(15)}`,
          lineHeight: rem(28),
          fontSize: rem(14),
        },
        default: {
          padding: `0 ${rem(30)}`,
          fontSize: rem(16),
          lineHeight: rem(38),
        },
        large: {
          padding: `0 ${rem(40)}`,
          lineHeight: rem(53),
          fontSize: rem(22),
        },
      },
      common: {
        fontWeight: '700',
        letterSpacing: '.4px',
      },
    },
    // form theme
    form: {
      regular: {
        states: {
          valid: {
            normal: {
              backgroundColor: colors.white,
              borderColor: colors.gray300,
              color: colors.gray700,
            },
            active: {
              backgroundColor: colors.white,
              borderColor: colors.primary500,
              color: colors.gray700,
            },
            disabled: {
              backgroundColor: colors.gray100,
              borderColor: colors.gray300,
              color: colors.gray600,
            },
          },
          invalid: {
            backgroundColor: colors.white,
            color: colors.danger500,
            borderColor: colors.danger500,
          },
        },
        common: {},
      },
      inline: {
        states: {
          valid: {
            normal: {
              backgroundColor: colors.transparent,
              borderColor: colors.transparent,
              color: colors.gray700,
            },
            hover: {
              backgroundColor: colors.transparent,
              borderColor: colors.gray300,
              color: colors.gray700,
            },
            active: {
              backgroundColor: colors.transparent,
              borderColor: colors.primary500,
              color: colors.gray700,
            },
            disabled: {
              backgroundColor: colors.gray100,
              borderColor: colors.gray300,
              color: colors.gray600,
            },
          },
          invalid: {
            backgroundColor: colors.transparent,
            color: colors.danger500,
            borderColor: colors.danger500,
          },
        },
        common: {},
      },
      // radio and checkbox theme
      misc: {
        states: {
          valid: {
            normal: {
              borderColor: colors.gray300,
              backgroundColor: colors.transparent,
            },
            active: {
              borderColor: colors.primary500,
              backgroundColor: colors.transparent,
            },
            checked: {
              borderColor: colors.transparent,
              backgroundColor: colors.primary500,
            },
          },
          invalid: {
            borderColor: colors.danger500,
            backgroundColor: colors.transparent,
          },
        },
        common: {},
      },
      // toggle theme
      toggle: {
        normal: {
          backgroundColor: colors.gray400,
        },
        checked: {
          backgroundColor: colors.primary500,
        },
      },
    },
    text: {
      sizings: {
        small: rem(14),
        medium: rem(16),
        large: rem(20),
      },
    },
    heading: {
      sizings: {
        h1: rem(36),
        h2: rem(32),
        h3: rem(28),
        h4: rem(24),
        h5: rem(20),
        h6: rem(18),
      },
    },
  },
};

export default theme;
