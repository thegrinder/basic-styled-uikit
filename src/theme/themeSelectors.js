export { getBoxBgColor, getBoxBorderColor } from '../components/ColorBox/colorBoxHelpers';
export { getHeadingSizing, getHeadingColor } from '../components/Headings/headingHelpers';
export { getTextSizing, getTextColor } from '../components/Text/textHelpers';
