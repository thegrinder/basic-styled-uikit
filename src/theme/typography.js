export const getBaseFontSize = ({ theme }) => theme.uiKit.typography.baseFontSize;

export const getBaseLineHeight = ({ theme }) => theme.uiKit.typography.baseLineHeight;

export const calcLineHeight = () => {};

const typography = {
  baseFontSize: 16,
  baseLineHeight: 1.5,
};

export default typography;
