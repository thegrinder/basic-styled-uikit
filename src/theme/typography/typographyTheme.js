const typography = {
  baseFontSize: 16,
  baseLineHeight: 1.5,
};

export default typography;
