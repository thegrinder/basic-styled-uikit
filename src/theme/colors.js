import { getColors } from './themeSelectors';

export const getColor = color => ({ theme }) => getColors(theme)[color];

const colors = {
  neutral0: '#fff',
  neutral100: '#f8f9fa',
  neutral400: '#ced4da',
  neutral600: '#868e96',
  neutral1000: '#161616',

  primary400: '#42A5F5',
  primary600: '#1E88E5',
  success400: '#66BB6A',
  success600: '#43A047',
  warning400: '#FFEE58',
  warning600: '#FDD835',
  danger400: '#FF8A65',
  danger600: '#F4511E',
};

export default colors;
