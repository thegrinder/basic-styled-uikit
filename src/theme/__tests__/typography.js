import theme from '../theme';
import typography, {
  getBaseFontSize,
  getBaseLineHeight,
  rem,
  calcLineHeight,
} from '../typography';

describe('typography', () => {
  describe('theme', () => {
    it('should be defined', () => {
      expect(typography).toBeDefined();
      expect(typography).toMatchSnapshot();
    });
  });

  describe('getBaseFontSize', () => {
    it('should return the correct base font size', () => {
      expect(getBaseFontSize(theme)).toEqual(
        theme.uiKit.typography.baseFontSize,
      );
    });
  });

  describe('getBaseLineHeight', () => {
    it('should return the correct base line height', () => {
      expect(getBaseLineHeight(theme)).toEqual(
        theme.uiKit.typography.baseLineHeight,
      );
    });
  });

  describe('rem', () => {
    it('should return the correct size in rem units with a default theme', () => {
      expect(rem(16)({ theme })).toEqual('1rem');
    });

    it('should return the correct size in rem units with a custom theme', () => {
      const customTheme = {
        uiKit: {
          ...theme.uiKit,
          typography: {
            ...theme.uiKit.typography,
            baseFontSize: 8,
          },
        },
      };
      expect(rem(16)({ theme: customTheme })).toEqual('2rem');
    });
  });

  describe('calcLineHeight', () => {
    it('should return the correct line height according to vertical rhythm and base line height', () => {
      expect(calcLineHeight('1rem', 1.5)).toEqual('1.5');
      expect(calcLineHeight('2rem', 1.5)).toEqual('1.5');
      expect(calcLineHeight('3rem', 1.5)).toEqual('1');
    });
  });
});
