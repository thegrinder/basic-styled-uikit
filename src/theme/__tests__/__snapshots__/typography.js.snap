// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`typography theme should be defined 1`] = `
Object {
  "baseFontSize": 16,
  "baseLineHeight": 1.5,
}
`;
